src/Games/NexusMods.Games.MountAndBlade2Bannerlord/BannerlordRunGameTool.cs

```diff
@@ -0,0 +1,83 @@
+public class BannerlordRunGameTool : RunGameTool<Bannerlord>
+{
+    private readonly ILogger<BannerlordRunGameTool> _logger;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly GameToolRunner _runner;
+
+    public BannerlordRunGameTool(IServiceProvider serviceProvider, Bannerlord game, GameToolRunner runner)
+        : base(serviceProvider, game)
+    {
+        _serviceProvider = serviceProvider;
+        _runner = runner;
+        _logger = serviceProvider.GetRequiredService<ILogger<BannerlordRunGameTool>>();
+    }
+
+    protected override bool UseShell { get; set; } = false;
+    
+    public override async Task Execute(Loadout.ReadOnly loadout, CancellationToken cancellationToken, string[]? commandLineArgs)
+    {
+        commandLineArgs ??= [];
+
+        // We need to 'inject' the current set of enabled modules in addition to any existing parameters.
+        // This way, external arguments specified by outside entities are preserved.
+        var args = await GetBannerlordExeCommandlineArgs(loadout, commandLineArgs, cancellationToken);
+        var install = loadout.InstallationInstance;
+        var exe = install.LocationsRegister[LocationId.Game];
+        if (install.Store != GameStore.XboxGamePass) { exe = exe/BinFolder/Win64Configuration/BannerlordExecutable; }
+        else { exe = exe/BinFolder/XboxConfiguration/BannerlordExecutable; }
+
+        var command = Cli.Wrap(exe.ToString())
+            .WithArguments(args)
+            .WithWorkingDirectory(exe.Parent.ToString());
+
+        // Note(sewer): We use the tool runner to execute an alternative process,
+        //              but because the UI expects `RunGameTool`, we override the `RunGameTool's` executor
+        //              with `GameToolRunner` implementation.
+        await _runner.ExecuteAsync(loadout, command, true, cancellationToken);
+    }
+
+    // TODO: Refactor Bannerlord.LauncherManager so we don't have to copy (some) of the commandline logic.
+    //       (Since LauncherManager has dependencies on code we don't need/want to provide).
+    private async Task<string[]> GetBannerlordExeCommandlineArgs(Loadout.ReadOnly loadout, string[] commandLineArgs, CancellationToken cancellationToken)
+    {
+        // Set the (automatic) load order.
+        // Copied from Bannerlord.LauncherManager
+        var manifestPipeline = Pipelines.GetManifestPipeline(_serviceProvider);
+        var modules = (await Helpers.GetAllManifestsAsync(_logger, loadout, manifestPipeline, cancellationToken).ToArrayAsync(cancellationToken))
+            .Select(x => x.Item2);
+        var sortedModules = AutoSort(Hack.GetDummyBaseGameModules()
+            .Concat(modules)).Select(x => x.Id).ToArray();
+        var loadOrderCli = sortedModules.Length > 0 ? $"_MODULES_*{string.Join("*", sortedModules)}*_MODULES_" : string.Empty;
+
+        // Add the new arguments
+        return commandLineArgs.Concat(["/singleplayer", loadOrderCli]).ToArray();
```

Comment:
So for now we are limited to Single player and we don't allow users to edit the load order?

I think it's fine to just get people something for now, but I would want to change this to start the Launcher and write the load order to file instead (by adding the load order file to Ignore entries and manually writing it out after Apply, since we don't have GeneratedFIles)

Reply:
We could theoretically do that once we have external changes available in the App.
For now, I've went with this simple approach as it's the most 'usable' we can make the game for the end users.

Booting directly into game with singleplayer is actually what the other managers do, so it's consistent with user expectation. That said, ideally I'd love to separate singleplayer and multiplayer out in the UI, and let the user boot the game in either mode directly from our end.